### ISSUE TITLE
Jasypt breaks properties injection with env variables

### ISSUE THREAD

**Issue by andreasreinhardt** [ulisesbocchio/jasypt-spring-boot#122]
jasypt spring boot version: 2.1.1
spring boot version: 2.1.2.RELEASE

How to reproduce:
Create a new Spring Boot Starter Web app and add a @ConfigurationProperties class with a String field. Add a @RequestMapping that returns this string. Start the app with the environment variable that sets this string. Calling this method will return the set value as expected.
If you add jasypt as additional dependency, the retunr value is null instead.

Workaround: jasypt.encryptor.proxyPropertySources=true

**Comment by ulisesbocchio** [comment]
Interesting, let me dig on it. Can you create a repo with your sample app?

**Comment by andreasreinhardt** [comment]
Here you are: https://github.com/andreasreinhardt/jasypt-environment-bug
By the way: I further found out that the problem only comes up when the properties contain dashes that I replace with underscores in the environment variable (which is valid for Spring relaxed properties binding). If I just remove the dash in the environment variable, it is working (see README).

**Comment by mbyrne00** [comment]
We've noticed the same thing - it's only failing for us when running a `@SpringBootTest` with `SpringExtension` runner. Worked fine when we started the actual server. Many thanks to @andreasreinhardt for his suggestion as setting that property in our `test` `application.yaml` fixed the problem for us.

**Comment by callmepills** [comment]
I am facing similar issue when trying to expose additional details via info actuator. In other non-Jasypt projects, I am able to define these environment variables as upper case with underscores in order to take advantage of the relaxed bindings.

```
INFO_TEST_PROPERTY   ## doesn't work
info.test.property   ## works
```

For now, I am just mapping the properties in `application.yml`:

```yml
info:
  test:
    property: ${INFO_TEST_PROPERTY}
```

**Comment by ulisesbocchio** [comment]
I dug a bit on this and according to the documentation (https://docs.spring.io/spring-boot/docs/current/reference/html/boot-features-external-config.html#boot-features-external-config-relaxed-binding) property: `acme.my-project.person.first-name` has the following relaxed bindings:

Property | Note
-- | --
acme.my-project.person.first-name | Kebab case, which is recommended for use in .properties and .yml files.
acme.myProject.person.firstName | Standard camel case syntax.
acme.my_project.person.first_name | Underscore notation, which is an alternative format for use in .properties and .yml files.
ACME_MYPROJECT_PERSON_FIRSTNAME | Upper case format, which is recommended when using system environment variables.

so in @andreasreinhardt case, for `spring.cloud.azure.eventhub.connection-string` it would be:

Property | Note
-- | --
spring.cloud.azure.eventhub.connection-string | Kebab case
spring.cloud.azure.eventhub.connectionString | Standard camel case syntax.
spring.cloud.azure.eventhub.connection_string | Underscore notation, which is an alternative format for use in .properties and .yml files.
SPRING_CLOUD_AZURE_EVENTHUB_CONNECTIONSTRING | Upper case format, which is recommended when using system environment variables.

Basically the ALL UPPERCASE binding just replaces `.` for `_`
